FogFog-iOS/FogFog-iOS/Networking/APIs/UserAPI.swift

```diff
@@ -22,8 +22,8 @@ extension UserAPI: FogAPI {
     
     var urlPath: String {
         switch self {
-        case .preferredMap(let userId):
-            return "/\(userId)/preffered-map"
+        case let .preferredMap(userId, _):
```

Comment:
음 습관적으로 작성하던 부분이었는데요. 앞에 한번만 작성해주면 전체 패턴에 다 매칭되어서 let을 여러번 작성해주지 않아도 된다는 점 때문에 이렇게 작성해왔던 것 같습니다.

```swift
case let .oneCase(a, b, c, d):
  // code
case .anotherCase(let a, let b, let c, let d):
  // code
```

이번 기회에 어떻게 작성하면 좋을까라는 생각해 서칭을 해보았는데요. [Google Swift Style Guide](https://google.github.io/swift/#switch-statements) 에서 생각해볼만한 내용이 있어서 인용해보겠습니다.

### Pattern Matching
The let and var keywords are placed individually in front of each element in a pattern that is being matched. The shorthand version of let/var that precedes and distributes across the entire pattern is forbidden because it can introduce unexpected behavior if a value being matched in a pattern is itself a variable.

```swift
enum DataPoint {
  case unlabeled(Int)
  case labeled(String, Int)
}

let label = "goodbye"

// `label` is treated as a value here because it is not preceded by `let`, so
// the pattern below matches only data points that have the label "goodbye".
// 1️⃣
switch DataPoint.labeled("hello", 100) {
case .labeled(label, let value):
  // ...
}

// Writing `let` before each individual binding clarifies that the intent is to
// introduce a new binding (shadowing the local variable within the case) rather
// than to match against the value of the local variable. Thus, this pattern
// matches data points with any string label.

// 2️⃣
switch DataPoint.labeled("hello", 100) {
case .labeled(let label, let value):
  // ...
}
```

In the example below, if the author’s intention was to match using the value of the label variable above, that has been lost because let distributes across the entire pattern and thus shadows the variable with a binding that applies to any string value:

// 3️⃣
```swift
switch DataPoint.labeled("hello", 100) {
case let .labeled(label, value):
  // ...
}
```

3️⃣번째 예시처럼 let이 전체에 걸쳐서 바인딩되는 경우 1️⃣번째 예시처럼 사용하려고 했던 코드 작성자의 원래 의도를 잃어버릴 수 있다는 이야기를 하고 있습니다. 결론을 내려보면 상황에 맞게 적절하게 사용해야겠지만, 개별적으로 let or var를 분배해서 작성해주는 것이 예상치 못한 동작을 방지할 수 있는 좋은 방법인 것 같다는 생각이 듭니다.